controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStream.java

```diff
@@ -0,0 +1,908 @@
+                                                             return r;
+                                                         }));
+    }
+
+    @Override
+    CompletableFuture<VersionedMetadata<StateRecord>> getStateData(boolean ignoreCached) {
+        return getMetadataTable()
+                .thenCompose(metadataTable -> {
+                    if (ignoreCached) {
+                        storeHelper.invalidateCache(getScope(), metadataTable, STATE_KEY);
+                    }
+
+                    return storeHelper.getCachedData(getScope(), metadataTable, STATE_KEY, StateRecord::fromBytes);
+                });
+    }
+
+    @Override
+    CompletableFuture<Void> createCommitTxnRecordIfAbsent(CommittingTransactionsRecord committingTxns) {
+        return getMetadataTable()
+                .thenCompose(metadataTable -> Futures.toVoid(storeHelper.addNewEntryIfAbsent(getScope(),
+                        metadataTable, COMMITTING_TRANSACTIONS_RECORD_KEY, committingTxns.toBytes())));
+    }
+
+    @Override
+    CompletableFuture<VersionedMetadata<CommittingTransactionsRecord>> getCommitTxnRecord() {
+        return getMetadataTable()
+                .thenCompose(metadataTable -> storeHelper.getEntry(getScope(), metadataTable, COMMITTING_TRANSACTIONS_RECORD_KEY, 
+                        CommittingTransactionsRecord::fromBytes));
+    }
+
+    @Override
+    CompletableFuture<Version> updateCommittingTxnRecord(VersionedMetadata<CommittingTransactionsRecord> update) {
+        return getMetadataTable()
+                .thenCompose(metadataTable -> storeHelper.updateEntry(getScope(), metadataTable, COMMITTING_TRANSACTIONS_RECORD_KEY, 
+                        update.getObject().toBytes(), update.getVersion()));
+    }
+
+    @Override
+    CompletableFuture<Void> createWaitingRequestNodeIfAbsent(String waitingRequestProcessor) {
+        return getMetadataTable()
+                .thenCompose(metadataTable -> Futures.toVoid(storeHelper.addNewEntryIfAbsent(getScope(),
+                        metadataTable, WAITING_REQUEST_PROCESSOR_PATH, waitingRequestProcessor.getBytes(StandardCharsets.UTF_8))));
+    }
+
+    @Override
+    CompletableFuture<String> getWaitingRequestNode() {
+        return getMetadataTable()
+                .thenCompose(metadataTable -> storeHelper.getEntry(getScope(), metadataTable, WAITING_REQUEST_PROCESSOR_PATH,
+                        x -> StandardCharsets.UTF_8.decode(ByteBuffer.wrap(x)).toString()))
+                .thenApply(VersionedMetadata::getObject);
+    }
+
+    @Override
+    CompletableFuture<Void> deleteWaitingRequestNode() {
+        return getMetadataTable()
+                .thenCompose(metadataTable -> storeHelper.removeEntry(getScope(), metadataTable, WAITING_REQUEST_PROCESSOR_PATH));
+    }
+
+    @Override
+    public void refresh() {
+        String id = idRef.get();
+        if (!Strings.isNullOrEmpty(id)) {
```

Comment:
This operation isn't really atomic. If it's needed, then we could achieve the same with an `updateAndGet` which guarantees atomicity.

Reply:
The atomicity isnt needed. ANd without affecting the buiness logic, i have changed the code to 
```
String id = idRef.getAndSet(null);
if (!Strings.isNullOrEmpty(id)) { 
// invalidate cached values
}
```